### ISSUE TITLE
Error is visible when WebSocket is unavailable

### ISSUE THREAD

**Issue by mmnosek** [Palindrom/Palindrom#184]
I'm at the airport trying to use Starcounter apps:
* http://warehouse.starcounter.com
* http://inside.starcounter.com

Both seems to work, but I see the error in the browser:

<img width="1424" alt="launchpad" src="https://user-images.githubusercontent.com/4943949/33673166-581bfeae-daac-11e7-866a-76b069468a6c.png">
<img width="1432" alt="warehouse" src="https://user-images.githubusercontent.com/4943949/33673177-5f477a14-daac-11e7-92e8-61c66516aafb.png">

It seems that WebSocket connection is unavailable (using airport WiFi) but an application shouldn't report any problem if it works. This is why there is `http` working when `WebSocket` unavailable, right? It is quite misleading for end users. Initially, I thought that everything is broken down.

Repro available at GDN airport ;)

@alshakero @tomalec @warpech

**Comment by alshakero** [comment]
It's exactly this issue https://github.com/Starcounter/Home/issues/318

**Comment by mmnosek** [comment]
> It's exactly this issue Starcounter/Home#318

I disagree. This issue is related, but it's about losing connection after some time of inactivity.

In my case, `WebSockets` are unavailable, but an application works - if works, a user shouldn't be aware of any problem. `WebSocket` is only a transportation layer. If not available, then we use `HTTP` - why then we tell the user about any error? The error should be displayed when a server is unavailable by any method. I expect smooth and nice UX for the enduser. 99,9% doesn't care if it's `websocket`, `http` or [Pigeon](https://en.wikipedia.org/wiki/IP_over_Avian_Carriers)

Reconnection should be tried from time to time, of course, but without notifying user (it works, even if a bit slower)

**Comment by warpech** [comment]
I agree with @mmnosek here. Disconnection of WebSocket should not be a fatal error for a Palindrom session.

If WebSocket is disconnected without an error in the closing message it simply means that the sync with the server is off. But the session might be completely valid.

We must display a "You are offline" warning and make sure we attempt to reconnect before the the session times out on server (20 minutes).

Only if:

- WebSocket is disconnected with an error message
- or HTTP request results in an error other than 502

@alshakero do you agree? Could you design a diagram about how reconnection should work in your opinion?
we can consider this to be a fatal error.

**Comment by alshakero** [comment]
> @alshakero do you agree? Could you design a diagram about how reconnection should work in your opinion?

Sure, I'll think about it, design it tomorrow and share here.

**Comment by mmnosek** [comment]
So, do I understand correctly, that as a result application will work on the airport (where `WebSocket` is unavailable) over `HTTP` and I will not see any kind of warning?

**Comment by alshakero** [comment]
Exactly, this is also necessary for ServiceWorkers to make any sense.

**Comment by mmnosek** [comment]
@alshakero any progress with this?

**Comment by alshakero** [comment]
@warpech how do you think we should prioritize this?

**Comment by warpech** [comment]
I believe it should be prioritized as in the "To do" column in https://github.com/orgs/Palindrom/projects/1 (top to bottom).

CSS framework has higher priority than this: https://github.com/Starcounter/RebelsLounge/issues/75

**Comment by mmnosek** [comment]
Great! Just don't want this issue to die. If it's in todo - great! I don't think the highest priority is needed. Thanks!
